#189 BUG-Add user by phone

KennanHoa opened the pull request:
#189 Supervisors can't add members to their group if the member has dashes in their phone number

-All non number are now removed before storing into the database so it doesn't conflict with when we search for the numbers.

KennanHoa committed:
#189 Add user by phone Bug

#189 Supervisors can't add members to their group if the member has dashes in their phone number

-All non number are now removed before storing into the database so it doesn't conflict with when we search for the numbers.

refatK commented on app/src/main/java/com/example/bigowlapp/activity/SignUpPageActivity.java:
```diff
@@ -62,7 +62,9 @@ protected void initialize() {
                 this.userPhone.requestFocus();
             } else {
-                signUpViewModel.createUser(email, pass, userPhone, firstName, lastName)
+                //String userPhone2 = userPhone.replaceFirst("(\\d{3})(\\d{3})(\\d+)", "($1) $2-$3");  //Format to be used?
```
You should probably remove the commented code

refatK commented on app/src/main/java/com/example/bigowlapp/activity/SignUpPageActivity.java:
```diff
@@ -62,7 +62,9 @@ protected void initialize() {
             } else {
-                signUpViewModel.createUser(email, pass, userPhone, firstName, lastName)
+                //String userPhone2 = userPhone.replaceFirst("(\\d{3})(\\d{3})(\\d+)", "($1) $2-$3");  //Format to be used?
+                String userPhone2 = userPhone.replaceAll("[^+0-9]", "");
```
rather than `userPhone2`, a more descriptive name of what you did might actually be better to indicate the fact its a number without the dashes.

Also this would be a [Rename Variable](https://refactoring.com/catalog/renameVariable.html) refactor, so please indicate that in the commit comment.

refatK reviewed (changes requested):
Just a couple of changes being requested, but seems like it should work correctly. Will test later!

KennanHoa committed:
Update SignUpPageActivity.java

ShawnGuerra-Bautista commented on app/src/main/java/com/example/bigowlapp/activity/SignUpPageActivity.java:
```diff
@@ -62,7 +62,9 @@ protected void initialize() {
             } else {
-                signUpViewModel.createUser(email, pass, userPhone, firstName, lastName)
+                //String userPhone2 = userPhone.replaceFirst("(\\d{3})(\\d{3})(\\d+)", "($1) $2-$3");  //Format to be used?
+                String formattedPhone = userPhone.replaceAll("[^+0-9]", "");
```
As the minimum requirement, the regex should account that a "+" sign should be in the beginning of the phone number, and that the phone number should end with a number.

If you feel adventurous and have the time, then consider using Google's phone number library https://github.com/google/libphonenumber. It can format number in a specific way. Moreover, it can actually compute whether the phone number is actually a real phone number with an actual area code, and country code. This is not mandatory, but consider it to make our code more robust.

KennanHoa replied:
Thanks! I looked it up but there's a huge amount of files so I'm a bit confused on how to use.
I did a quick fix for now

ShawnGuerra-Bautista reviewed (changes requested):
Overall, the code changes are straightforward and good.

The only issue is regarding the Regex.

refatK reviewed (approved):
Thanks for addressing my request.  I've also now ran the code by registering a user and it worked as I expected. I think the only issue is related what @ShawnGuerra-Bautista mentioned in his review.

registering number with dashes and spaces:
![image](https://user-images.githubusercontent.com/31963426/108581970-6499e900-72fe-11eb-9cca-7b2df7f0597f.png)


result in database with cleaned format:
![image](https://user-images.githubusercontent.com/31963426/108581962-5c41ae00-72fe-11eb-9af7-b0c119457799.png)

KhangZhengNg reviewed (dismissed):
Thanks for fixing bug regarding regex in profile editting. Try to fix exactly what Shawn suggested and you can merge!

KennanHoa committed:
#189 Improved phone format

- Remove comment
- Rename variable
- Improve format (If contains '+' --> required to start with it. Phone also need to end with a number.)

KennanHoa committed:
Merge branch 'staging' into #189-BUG-AddUserByPhone

KennanHoa committed:
#189 Changed format using LibPhoneNumber

- Using Google repository for phone format and error handling

KennanHoa committed:
Update EditProfileViewTest.java
